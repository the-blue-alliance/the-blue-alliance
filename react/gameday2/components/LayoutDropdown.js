import React, { PropTypes } from 'react'
import LayoutDropdownItem from './LayoutDropdownItem'

export default React.createClass({
  propTypes: {
    setLayout: PropTypes.func.isRequired,
  },
  layoutSelected(layoutId) {
    this.props.setLayout(layoutId)
  },
  render() {
    return (
      <li className="dropdown">
        <a href="#" className="dropdown-toggle" data-toggle="dropdown">Set Layout <b className="caret" /></a>
        <ul className="dropdown-menu">
          <LayoutDropdownItem handleClick={this.layoutSelected} layoutId={0}>Single View</LayoutDropdownItem>
          <LayoutDropdownItem handleClick={this.layoutSelected} layoutId={1}>Split View</LayoutDropdownItem>
          <LayoutDropdownItem handleClick={this.layoutSelected} layoutId={2}>"1+2" View</LayoutDropdownItem>
          <LayoutDropdownItem handleClick={this.layoutSelected} layoutId={3}>Quad View</LayoutDropdownItem>
          <LayoutDropdownItem handleClick={this.layoutSelected} layoutId={4}>"1+3" View</LayoutDropdownItem>
          <LayoutDropdownItem handleClick={this.layoutSelected} layoutId={5}>"1+4" View</LayoutDropdownItem>
          <LayoutDropdownItem handleClick={this.layoutSelected} layoutId={6}>Hex View</LayoutDropdownItem>
          <LayoutDropdownItem handleClick={this.layoutSelected} layoutId={7}>Octo-View</LayoutDropdownItem>
          <LayoutDropdownItem handleClick={this.layoutSelected} layoutId={8}>Nona-View</LayoutDropdownItem>
        </ul>
      </li>
    )
  },
})
