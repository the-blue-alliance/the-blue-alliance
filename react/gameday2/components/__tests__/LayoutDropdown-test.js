jest.unmock('../LayoutDropdown')

import React from 'react'
import LayoutDropdown from '../LayoutDropdown'
import renderer from 'react-test-renderer'

describe('LayoutDropdown', () => {
  it('renders the correct contents', () => {
    const dropdown = renderer.create(
      <LayoutDropdown setLayout={() => {}} />
    ).toJSON()
    expect(dropdown).toMatchSnapshot()
  })
})
