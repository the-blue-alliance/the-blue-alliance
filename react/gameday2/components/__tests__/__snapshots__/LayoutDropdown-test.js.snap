exports[`LayoutDropdown renders the correct contents 1`] = `
<li
  className="dropdown">
  <a
    className="dropdown-toggle"
    data-toggle="dropdown"
    href="#">
    Set Layout 
    <b
      className="caret" />
  </a>
  <ul
    className="dropdown-menu" />
</li>
`;
